Makefile

```diff
@@ -1,46 +1,91 @@
 ## Copyright (c) 2015 Ahmad Fatoum
+
+# define commands
 CROSS_COMPILE ?= C:/CSLite/bin/arm-none-linux-gnueabi-
 PREFIX ?= $(CROSS_COMPILE)
-DESTDIR ?= $(CURDIR)
 CC = $(PREFIX)gcc
 AR = $(PREFIX)ar
 SED = sed
 MKDIR = mkdir -p
 INSTALL = cp
 RM = rm -rf
+
+# define directories
+# - for build
+DEPDIR := .deps
+OBJDIR := .objs
+# - for install
+DESTDIR    ?= $(CURDIR)/installed
+INCLUDEDIR ?= $(DESTDIR)/include
+LIBDIR     ?= $(DESTDIR)/lib
+
+# define files
 SRCS = $(wildcard API/*.c contrib/**/*.c)
-OBJS = $(patsubst %.c,%.o,$(SRCS))
+OBJS = $(patsubst %.c,$(OBJDIR)/%.o,$(SRCS))
+DEPFILES = $(patsubst %.c,$(DEPDIR)/%.d,$(SRCS))
+
+# define flags
+DEPFLAGS = -MT $@ -MMD -MP -MF $(DEPDIR)/$*.d
 override CFLAGS += -std=c99
 override CFLAGS += -fno-strict-aliasing -fwrapv
 override CFLAGS += -Wall -Wextra -Wpointer-sign -Wno-unused-parameter
 override CFLAGS += -D_GNU_SOURCE=1
 
-.DEFAULT: libev3api.a
+# logging control
+ifeq ($(VERBOSE),)

```

Comment:
Nitpick: `ifeq ($(VERBOSE)$(V),)` for the Autotools impaired..

Reply:
That's an elegant way of doing an OR, thanks! Fixed in https://github.com/c4ev3/EV3-API/pull/49/commits/e173fc611889534aa8f23592fe4870df96a77c05